### PR TITLE
[Android] Use c++11 range for loops

### PR DESCRIPTION
**Pull request by enen92**
## Description
Follow up to https://github.com/xbmc/xbmc/pull/19232 implements the changes requested in https://github.com/xbmc/xbmc/pull/19291#discussion_r583641084

Decided to adjust the other loop over m_activityResultEvents too (the source of the original problem).

## Motivation and Context
New standards

## How Has This Been Tested?
Untested. I'll trigger a test build from jenkins and test tonight.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/platform/android/activity/XBMCApp.cpp [hunk 1]
```diff
@@ -1251,13 +1251,13 @@ int CXBMCApp::WaitForActivityResult(const CJNIIntent &intent, int requestCode, C
...
+  // delete from m_activityResultEvents before deleting the event
   CSingleLock lock(m_activityResultMutex);
-  for (auto it = m_activityResultEvents.begin(); it != m_activityResultEvents.end(); ++it)
+  for (const auto& activityResultEvent : m_activityResultEvents)
   {
-    if ((*it)->GetRequestCode() == requestCode)
+    if (activityResultEvent->GetRequestCode() == requestCode)
     {
-      m_activityResultEvents.erase(it);
+      m_activityResultEvents.erase(activityResultEvent);
```
Comment (ksooo) [inline xbmc/platform/android/activity/XBMCApp.cpp:1260]: Hmm, isn't erase with a key_type way slower than erase with an iterator?

https://www.cplusplus.com/reference/map/map/erase/

Complexity
For the first version (erase(position)), amortized constant.
For the second version (erase(val)), logarithmic in container size.

**Review by ksooo — COMMENTED** [review]
Not sure this change is a good idea.

File: xbmc/platform/android/activity/XBMCApp.cpp (hunk 1 above)
Comment (howie-f) [reply xbmc/platform/android/activity/XBMCApp.cpp:1260]: mh, i assumed we get an iterator returned by the range based loop, now looking at it again i'm figuring if an erase with a const ref is possible at all... is that runtime tested, @enen92 ?
sorry for any confusion

Comment (enen92) [reply xbmc/platform/android/activity/XBMCApp.cpp:1260]: Not runtime tested, building in jenkins as explained in the pr description

Comment (ksooo) [reply xbmc/platform/android/activity/XBMCApp.cpp:1260]: That will build, but as i said, it has runtime performance implications.

Comment (enen92) [reply xbmc/platform/android/activity/XBMCApp.cpp:1260]: @ksooo I also agree the style changes are not worth a penalty on performance (although the vector is/will always be small). Conclusion: close this one and reopen the original backport?

**Pull request closed without merging**